src/event/model/set_uuid_state.rs

```diff
@@ -22,3 +26,96 @@ impl From<AbstractEvent> for SetUuidState {
         }
     }
 }
+
+pub struct SetUuidStateEventPayload {
+    __typename: EventType,
+    raw_typename: String,
+    actor_id: i32,
+    object_id: i32,
+    trashed: bool,
+    instance: String,
+}
+
+impl SetUuidStateEventPayload {
+    pub fn new(trashed: bool, actor_id: i32, object_id: i32, instance: &str) -> Self {
+        // TODO: handle datetime yourself
+
+        let raw_typename = if trashed {
+            "uuid/trash".to_string()
+        } else {
+            "uuid/restore".to_string()
+        };
+
+        SetUuidStateEventPayload {
+            __typename: EventType::SetThreadState,
+            raw_typename,
+            actor_id,
+            object_id,
+            trashed,
+            instance: instance.to_string(),
+        }
+    }
+
+    pub async fn save<'a, E>(&self, executor: E) -> Result<Event, EventError>
+    where
+        E: Executor<'a>,
+    {
+        let mut transaction = executor
+            .begin()
+            .await
+            .map_err(|inner| EventError::DatabaseError { inner })?;
+        sqlx::query!(
+            r#"
+                INSERT INTO event_log (actor_id, event_id, uuid_id, instance_id)
+                    SELECT ?, e.id, ?, i.id
+                    FROM event e
+                    JOIN instance i
+                    WHERE e.name = ? AND i.subdomain = ?
+            "#,
+            self.actor_id,
+            self.object_id,
+            self.raw_typename,
+            self.instance
+        )
+        .execute(&mut transaction)
+        .await
+        .map_err(|inner| EventError::DatabaseError { inner })?;
+        let value = sqlx::query!(r#"SELECT LAST_INSERT_ID() as id"#)
+            .fetch_one(&mut transaction)
+            .await
+            .map_err(|inner| EventError::DatabaseError { inner })?;
+        let event = Event::fetch(value.id as i32, &mut transaction).await?;
+        transaction
+            .commit()
+            .await
+            .map_err(|inner| EventError::DatabaseError { inner })?;
+
+        Ok(event)
```

Comment:
This whole function is a transaction (or possibly a sub transaction when the passed executor is already a transaction like in the tests). So: the mutations and the concluding query will only be executed as a group.